src/browser/index.ts

```diff
@@ -6,12 +6,14 @@ import {assertEnvSupport} from './support'
 export function groqStore(config: Config): GroqStore {
   assertEnvSupport()
 
-  if (config.token) {
+  const EventSource = config.EventSource ?? window.EventSource
+
+  if (config.token && EventSource === window.EventSource) {
     throw new Error('`token` option not currently supported in browser')
```

Comment:
```suggestion
    throw new Error('When`token` option is used, `EventSource` option must also be provided. EventSource cannot be `window.EventSource`, as it does not support passing a token.')
```
or something like that?

Reply:
Could you also update the README.md to reflect the new api?

Reply:
Like this? https://github.com/sanity-io/groq-store/pull/9/commits/9052182075464548e2d80c9169e9a234e6ca997b

I kept the `EventSource` parameter commented out, so it will still work as is with copy-paste